client/src/app/payroll/payroll-list/payroll-list.component.ts

```diff
@@ -1,20 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
+
 import { faSkull, faSmileBeam } from '@fortawesome/free-solid-svg-icons';
-import { DataService } from 'src/app/data/data-service';
-import { Payroll } from 'src/app/data/state/payroll-model';
+
+import { PayrollListQuery } from './state/payroll-list.query';
+import { PayrollListService } from './state/payroll-list.service';
 
 @Component({
   selector: 'app-payroll-list',
   templateUrl: './payroll-list.component.html',
   styleUrls: ['./payroll-list.component.scss'],
 })
-export class PayrollListComponent {
+export class PayrollListComponent implements OnInit {
   readonly faSkull = faSkull;
   readonly faSmileBeam = faSmileBeam;
+  readonly payrolls = this.query.selectAll();
+  private employeeDepartment: string;
+
+  @Input()
+  set department(department: string) {
+    const employeeDepartment = department && department.trim();
```

Comment:
You can rearrange this to get rid of the local variable:

```typescript
this.employeeDepartment = department?.trim() ?? '';

this.service.getPayrolls(employeeDepartment);
```

Reply:
Nice suggestion!